infrahub_sdk/checks.py

```diff
@@ -81,6 +82,11 @@ def client(self, value: InfrahubClient) -> None:
     @classmethod
     async def init(cls, client: Optional[InfrahubClient] = None, *args: Any, **kwargs: Any) -> InfrahubCheck:
         """Async init method, If an existing InfrahubClient client hasn't been provided, one will be created automatically."""
+        warnings.warn(
+            "InfrahubCheck.init has been deprecated and will be removed in the version in Infrahub SDK 1.1.0",
```

Comment:
This code will ship for the first time with 1.1 so this can't be accurate
To be clean let's say 2.0

Reply:
I'd like to get rid of the `InfrahubClient()` call below as it requires us to specifically import InfrahubClient. Are you fine with saying "after 1.1.0" instead? Alternatively we can change it to 2.0 and use importlib or similar here until that point. I don't think anyone outside of infrahubctl or the git-agent ever used this.

Reply:
I've pushed an update to schedule this method to be removed in SDK 2.0 instead and at the same type we change how InfrahubClient is used within that module in order to change the InfrahubCheck module to follow the same logic as used for Transforms and generators in #128.